frame/contracts/src/wasm/runtime.rs

```diff
@@ -1847,4 +1881,45 @@ define_env!(Env, <E: Ext>,
 			Err(_) => Ok(ReturnCode::EcdsaRecoverFailed),
 		}
 	},
+
+	// Execute code in the context of the current contract.
+	//
```

Comment:
Need some more docs here describing the following:
- The code is executed in the same storage transaction as the caller
- Reentrancy protection is always **disabled** because you need to trust the callee anyways (please also add this to the docs of `CallFlags::ALLOW_REENTER`.